### PR TITLE
Gradle implementation for Visual Studio 2013

### PR DESCRIPTION
**Pull request by markm11**
I have seen some request for gradle support.  This gradle setup has the ability to build x86 as well as x64.  I needed x64 on a application I am developing with gradle.

### CODE REVIEW

**Comment by markm11** [comment]
I did use a different e-mail when I signed the CLA then I use here on Git.   Sorry about that.

File: build.gradle [hunk 1]
```diff
@@ -1,66 +1,135 @@
...
-          else{
-            linker.args "/MACHINE:X64"
-          }
-          if(buildType == buildTypes.Debug){
-            cppCompiler.define "_DEBUG"
-            cppCompiler.args "/Gm", "/Od", "/sdl", "/RTC1", "/MTd"
-            if(targetPlatform == platforms.x86){
-              cppCompiler.args "/ZI"
+  if(_MSC_VER == 1900){
+		VS_2015_INCLUDE_DIR = "C:/Program Files (x86)/Windows Kits/10/Include/10.0.10586.0/ucrt"
```
Comment (davidzchen) [inline build.gradle]: Can we keep the indentation style consistent (2 spaces rather than tab)?

File: build.gradle [hunk 2]
```diff
@@ -6,130 +6,130 @@ def VS_2015_LIB_DIR = ""
 
 allprojects{
```
Comment (davidzchen) [inline build.gradle]: Thanks for making the formatting changes! However, some of the indentation and brace placement is still off (see comments below).

For here and elsewhere, make sure there is a space before the brace.

File: build.gradle [hunk 3]
```diff
@@ -6,130 +6,130 @@ def VS_2015_LIB_DIR = ""
...
   if(_MSC_VER == 1900){
```
Comment (davidzchen) [inline build.gradle]: Space after if and before the brace

File: build.gradle [hunk 4]
```diff
@@ -6,130 +6,130 @@ def VS_2015_LIB_DIR = ""
...
-						projectFile.withXml
-							{
-							asNode().@ToolsVersion = "12.0"
-							asNode().PropertyGroup.findAll({it.@Label == 'Configuration'}).each{ configNode ->
-							  configNode.appendNode("PlatformToolset","v120")
-							  }
-						  }
+    VS_2015_INCLUDE_DIR = "C:/Program Files (x86)/Windows Kits/10/Include/10.0.10586.0/ucrt"
+    VS_2015_LIB_DIR = "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.10586.0/ucrt"
+    }
```
Comment (davidzchen) [inline build.gradle]: Closing brace should be on the previous indentation. Same elsewhere

File: build.gradle [hunk 5]
```diff
@@ -6,130 +6,130 @@ def VS_2015_LIB_DIR = ""
...
-		buildTypes{
-			Debug
-			Release
-			}
+          else{
+            installDir "C:/Program Files (x86)/Microsoft Visual Studio 12.0"
+            }
+          }
+        }
+      else{
```
Comment (davidzchen) [inline build.gradle]: Indentation here is really off. Also, `else` should be on the same line as the closing brace.

File: build.gradle [hunk 6]
```diff
@@ -6,130 +6,130 @@ def VS_2015_LIB_DIR = ""
...
-				if(toolChain in Gcc){
-					if(buildType == buildTypes.Debug){
-						cppCompiler.args "-g"
-						}
-					}
-				}
-			}
-		}
-	}
+    platforms{
```
Comment (davidzchen) [inline build.gradle]: Indentation in this section seems to have been thrown off by not properly de-indenting closing braces.

Should be:

```groovy
    platforms {
      x86 {
        architecture "x86"
      }
      x64 {
        architecture "x86_64"
      }
    }
    ...
```

and et cetera.

**Comment by davidzchen** [comment]
Style looks good to me. Thanks!

Can one of the project owners review this change?

**Comment by gennadiycivil** [comment]
This is sadly, out of date and there are merge conflicts. My apologies but the best way forward seems to be closing this PR. Please review the conflicts and resubmit when ready

**Pull request closed without merging**